src/company/company.service.ts

```diff
@@ -0,0 +1,105 @@
+import { Inject, Injectable, NotFoundException } from "@nestjs/common";
+import { UpdateCompanyDto } from "./dtos/updateCompany.dto";
+import { GetCompanyQueryDto } from "./dtos/companyGetQuery.dto";
+import { FindOptions, Transaction } from "sequelize";
+import { CompanyModel, JobModel } from "../db/models";
+import { parseFilter, parseOrder, parsePagesize } from "../utils";
+import { COMPANY_DAO } from "../constants"; // Assuming you have these DTO
+
+@Injectable()
+export class CompanyService {
+  constructor(@Inject(COMPANY_DAO) private companyRepo: typeof CompanyModel) {}
+
+  async getAllCompanies(where: GetCompanyQueryDto) {
+    const findOptions: FindOptions<CompanyModel> = {
+      include: [
+        {
+          model: JobModel,
+          as: "job",
+        },
+      ],
+    };
+
+    // Add page size options
+    const pageOptions = parsePagesize(where);
+    Object.assign(findOptions, pageOptions);
+    // Apply filter
+    parseFilter(findOptions, where.filterBy || {});
+    // Apply order
+    findOptions.order = parseOrder(where.orderBy || {});
+
+    const companies = await this.companyRepo.findAll(findOptions);
+
+    return companies.map((company) => company.get({ plain: true }));
+  }
+
+  async createCompany(body, t: Transaction) {
+    const createdIds = [];
+
+    try {
```

Comment:
1. Use BulkCreate wherever possible to parallelize things. If not, create promises and do `await Promise.all([])`
2. don't need to rollback or commit since you are already using `TransactionInterceptor`
3. don't need try-catch anywhere since we already have a global try-catch (in Exception filter)